app/views/RoomMembersView/index.js

```diff
@@ -114,6 +114,13 @@ class RoomMembersView extends React.Component {
 		}
 	}
 
+	componentDidUpdate(_, prevState) {
```

Comment:
This is getting called because `subscription` is being updated and not because of the user's roles being changed.
In order to do that, we should listen to changes on `login.user.roles`